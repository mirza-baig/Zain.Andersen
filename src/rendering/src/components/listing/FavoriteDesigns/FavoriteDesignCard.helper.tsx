import React, { useState } from 'react';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { Headline } from 'src/helpers/Headline';
import { ImageToggleWrapper } from 'src/helpers/ImageToggleWrapper';
import { getButtonGroupData, getImageToggleData } from './FavoriteDesigns.helper';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { getBreakpoint, useCurrentScreenType } from 'lib/utils/get-screen-type';
import { DesignProps } from './FavoriteDesignsTypes.helper';
import { getLocalStorageItem, setLocalStorageItem } from 'lib/utils/client-storage-utils';
import classNames from 'classnames';
import { BodyCopy } from 'src/helpers/BodyCopy';

const FavoriteDesignCard = (props: DesignProps) => {
  const [isFavoriteDesign, setIsFavoriteDesign] = useState(true);
  const { currentScreenWidth } = useCurrentScreenType();

  const updateFavoriteStatus = () => {
    let designs = getLocalStorageItem('aw_favoritedesigns') as Array<DesignProps>;
    if (isFavoriteDesign) {
      designs = designs.filter(
        (designItem) => JSON.stringify(designItem) !== JSON.stringify(props)
      );
    } else {
      designs.push(props);
    }
    setLocalStorageItem('aw_favoritedesigns', designs);

    const countStorageEvent = new StorageEvent('storage', {
      key: 'aw_favoritedesigns',
      newValue: JSON.stringify(designs),
    });
    // Dispatch the event
    window.dispatchEvent(countStorageEvent);

    setIsFavoriteDesign(!isFavoriteDesign);
  };
  if (!props.requestAQuoteUrl) {
    const newProps = {
      ...props, // Copy old values
      requestAQuoteUrl: props.url ? props.productDetailUrl + '#request_a_quote' : '',
    };

    props = newProps;
  }
  return (
    <div className="grid-rows-auto relative grid grid-cols-12 gap-y-0 p-s py-s ml:max-w-screen-lg ml:grid-flow-row-dense ml:grid-cols-12 ml:gap-s ml:px-0 lg:mx-auto">
      {/* Favorite icon */}
      <div className="absolute top-0 right-0 print:hidden" onClick={() => updateFavoriteStatus()}>
        <div
          className={classNames(
            'absolute right-0 top-0 inline h-0 w-0 cursor-pointer border-t-0 border-l-0 border-r-[60px] border-b-[60px] border-solid transition-[border-color]  duration-500 ease-[ease]',
            isFavoriteDesign
              ? 'border-[transparent_#F26924_transparent_transparent]'
              : 'border-[transparent_#C4BFB6_transparent_transparent]'
          )}
        >
          <span className="absolute -right-[51px] top-[11px]">
            <svg
              role="img"
              width="20"
              height="20"
              viewBox="0 0 18 16"
              fill="none"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>Favorite Icon</title>
              <path
                d="M9.26884 16L8.37389 15.1924C6.83137 13.7808 5.55806 12.5621 4.55397 11.5362C3.54988 10.5102 2.74951 9.59345 2.15288 8.78581C1.55624 7.97817 1.13787 7.24693 0.897761 6.59209C0.657652 5.93724 0.537598 5.27513 0.537598 4.60573C0.537598 3.29604 0.977798 2.201 1.8582 1.3206C2.7386 0.4402 3.82637 0 5.1215 0C5.95097 0 6.71859 0.196453 7.42436 0.589359C8.13014 0.982265 8.74496 1.5498 9.26884 2.29195C9.88003 1.50614 10.5276 0.927694 11.2115 0.556617C11.8955 0.185539 12.6304 0 13.4162 0C14.7113 0 15.7991 0.4402 16.6795 1.3206C17.5599 2.201 18.0001 3.29604 18.0001 4.60573C18.0001 5.27513 17.88 5.93724 17.6399 6.59209C17.3998 7.24693 16.9814 7.97817 16.3848 8.78581C15.7882 9.59345 14.9878 10.5102 13.9837 11.5362C12.9796 12.5621 11.7063 13.7808 10.1638 15.1924L9.26884 16Z"
                fill="white"
              ></path>
            </svg>
          </span>
        </div>
      </div>
      {currentScreenWidth >= getBreakpoint('ml') && (
        <div className="col-span-12 ml:col-span-4 [@media_print]:col-span-12">
          <ImageToggleWrapper {...getImageToggleData(props)} />
        </div>
      )}
      <div className="col-span-12 ml:col-span-7 [@media_print]:col-span-12 [@media_print]:mx-m">
        <div className="mb-m flex flex-col justify-between border-b border-gray px-xxs ml:mb-0 ml:flex-row ml:px-0">
          <div className="">
            <Headline
              classes="font-heavy text-xs ml:text-s pb-xxs"
              fields={{
                headlineText: { value: props.productName || props.seriesCategory },
              }}
            />
            <BodyCopy
              fields={{ body: { value: props.productShortDescription } || '' }}
              classes=""
            />
          </div>

          <div className="a2a_kit a2a_kit_size_32 a2a_default_style mb-xxs cursor-pointer print:hidden ml:mb-0">
            <a
              className="a2a_dd flex items-center text-darkprimary ml:justify-center"
              href="https://www.addtoany.com/share"
            >
              <SvgIcon icon="share" />
              <span className="ml-xxs">Share</span>
            </a>
          </div>
        </div>
        {currentScreenWidth <= getBreakpoint('ml') && (
          <div className="col-span-12 ml:col-span-4">
            <ImageToggleWrapper {...getImageToggleData(props)} />
          </div>
        )}
        <div className="pb-m ml:py-m">
          <Headline
            classes="font-heavy text-sm-xs ml:text-xs pb-s ml:pb-m"
            fields={{
              headlineText: { value: 'Design Summary' },
            }}
          />
          <div className="flex flex-col flex-wrap ml:flex-row">
            {props.selections.map((item: { title?: string; value?: string }) => {
              return (
                item.title !== 'Product ID#' && (
                  <div className="flex items-center ml:basis-1/2 ml:pb-xxs">
                    <div className="basis-1/2 font-sans text-small font-heavy ml:text-xxs ml:leading-none">
                      {item.title}
                    </div>
                    <div className="basis-1/2 pr-s text-small ml:text-body">{item.value}</div>
                  </div>
                )
              );
            })}
          </div>
        </div>
        <ButtonGroup
          classes={{
            wrapper: 'flex-col ml:items-center',
            cta1Classes: 'mr-2',
            cta2Classes: 'my-s ml:my-0 !ml-0',
          }}
          {...getButtonGroupData(props)}
        />
      </div>
    </div>
  );
};

export default FavoriteDesignCard;
