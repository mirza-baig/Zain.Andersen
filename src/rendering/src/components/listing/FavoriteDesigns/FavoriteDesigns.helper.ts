import { ButtonGroupProps } from 'src/helpers/ButtonGroup';
import { ImageToggleWrapperProps } from 'src/helpers/ImageToggleWrapper/ImageToggleWrapper';
import { DesignProps } from './FavoriteDesignsTypes.helper';

export const getImageToggleData = (props: DesignProps): ImageToggleWrapperProps => {
  return {
    fields: {
      primaryImage: {
        value: {
          src: props.interiorImage?.src,
          width: 387,
          height: 387,
        },
      },
      primaryImageMobile: {
        value: {
          src: props.interiorImage?.src,
          width: 387,
          height: 387,
        },
      },
      secondaryImage: {
        value: {
          src: props.exteriorImage?.src,
          width: 387,
          height: 387,
        },
      },
      secondaryImageMobile: {
        value: {
          src: props.exteriorImage?.src,
          width: 387,
          height: 387,
        },
      },
    },
  };
};

export const getButtonGroupData = (props: DesignProps): Partial<ButtonGroupProps> => {
  return {
    fields: {
      cta1Style: {
        name: '',
        id: '',
        url: '',
        fields: {
          Value: {
            value: 'primary',
          },
        },
      },
      cta1Link: {
        value: {
          href: decodeURIComponent(props.requestAQuoteUrl),
          linktype: 'internal',
          text: 'Get a quote',
          querystring: '',
          target: '',
          id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
        },
      },
      cta1Icon: {
        id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
        url: '',
        name: 'Arrow',
        displayName: 'Arrow',
        fields: {
          Value: {
            value: 'arrow',
          },
        },
        templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
        templateName: 'Enum',
      },
      cta1ModalLinkText: { value: '' },
      cta1AriaLabel: {
        value: '',
      },
      cta2Style: {
        name: '',
        url: '',
        id: '',
        fields: {
          Value: {
            value: 'link',
          },
        },
      },
      cta2Link: {
        value: {
          href: props.url,
          linktype: 'internal',
          text: 'Edit design selections',
          querystring: '',
          target: '',
          id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
        },
      },
      cta2Icon: {
        id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
        url: '',
        name: 'Arrow',
        displayName: 'Arrow',
        fields: {
          Value: {
            value: 'arrow',
          },
        },
        templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
        templateName: 'Enum',
      },
      cta2ModalLinkText: { value: '' },
      cta2AriaLabel: {
        value: '',
      },
      cta3Style: {
        name: '',
        url: '',
        id: '',
        fields: {
          Value: {
            value: 'link',
          },
        },
      },
      cta3Link: {
        value: {},
      },
      cta3Icon: {
        id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
        url: '',
        name: 'Arrow',
        displayName: 'Arrow',
        fields: {
          Value: {
            value: 'arrow',
          },
        },
        templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
        templateName: 'Enum',
      },
      cta3ModalLinkText: { value: '' },
      cta3AriaLabel: {
        value: '',
      },
    },
  };
};

export const shareServicesToExclude = [
  'amazon_wish_list',
  'aol_mail',
  'balatarin',
  'bibsonomy',
  'bitty_browser',
  'blinklist',
  'blogger',
  'blogmarks',
  'bookmarks_fr',
  'box_net',
  'buffer',
  'care2_news',
  'citeulike',
  'copy_link',
  'design_float',
  'diary_ru',
  'diaspora',
  'digg',
  'diigo',
  'douban',
  'draugiem',
  'dzone',
  'evernote',
  'facebook_messenger',
  'fark',
  'flipboard',
  'folkd',
  'google_bookmarks',
  'google_classroom',
  'hacker_news',
  'hatena',
  'houzz',
  'instapaper',
  'kakao',
  'kik',
  'kindle_it',
  'known',
  'line',
  'linkedin',
  'livejournal',
  'mail_ru',
  'mastodon',
  'mendeley',
  'meneame',
  'mewe',
  'mix',
  'mixi',
  'myspace',
  'netvouz',
  'odnoklassniki',
  'papaly',
  'pinboard',
  'plurk',
  'pocket',
  'print',
  'printfriendly',
  'protopage_bookmarks',
  'pusha',
  'qzone',
  'reddit',
  'rediff',
  'refind',
  'renren',
  'sina_weibo',
  'sitejot',
  'skype',
  'slashdot',
  'sms',
  'stocktwits',
  'svejo',
  'symbaloo_bookmarks',
  'telegram',
  'threema',
  'trello',
  'tuenti',
  'tumblr',
  'twiddla',
  'twitter',
  'typepad_post',
  'viadeo',
  'viber',
  'vk',
  'wanelo',
  'wechat',
  'whatsapp',
  'wordpress',
  'wykop',
  'xing',
  'yoolink',
  'yummly',
];
