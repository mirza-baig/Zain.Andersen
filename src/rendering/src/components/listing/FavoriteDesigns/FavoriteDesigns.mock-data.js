const defaultData = {
  rendering: {
    dataSource: 'sampledatasource',
    componentName: 'FavoriteDesigns',
  },
  fields: {
    noResultsHeadline: {
      value: 'Looks like you have no saved designs.',
    },
    noResultsBodyCopy: {
      value: "Let's favorite some things.",
    },
    card1Headline: { value: 'Lorem ipsum windows' },
    card1Subheadline: { value: 'Subheading text' },
    card2Headline: { value: 'Lorem ipsum doors' },
    card2Subheadline: { value: 'Subheading text' },
    componentSpacing: null,
    sectionId: { value: '' },
    headlineLevel: {
      id: '7f10288a-beb1-4d95-a139-32df40094441',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3',
      name: 'Heading 3',
      displayName: 'Heading 3',
      fields: { Value: { value: 'h3' } },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    headlineText: { value: 'Favorite Designs' },
    primaryImageMobileFocusArea: null,
    primaryImage: {
      value: {
        src: '/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Big-Doors-591-591.jpg?h=591&iar=0&w=591&hash=34B1FA75EB0F6C48783872FFEE2E97AB',
        alt: '591*591',
        width: '591',
        height: '591',
      },
    },
    primaryImageMobile: {
      value: {
        src: '/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Big-Doors-900x900.jpg?h=900&iar=0&w=900&hash=EAF4536D3DF642BACC89A8DD309541D4',
        alt: 'Big-Doors-900x900',
        width: '900',
        height: '900',
      },
    },
    secondaryImage: {
      value: {
        src: '/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Brian-Test/photo-1481467804160-3cdf91aa1ac4.jpg?h=1200&iar=0&w=1200&hash=7AAB1557107CA20AFFF24F95A2CF1E9B',
        alt: 'Window',
        width: '1200',
        height: '1200',
      },
    },
    secondaryImageMobileFocusArea: null,
    secondaryImageMobile: {
      value: {
        src: '/-/media/Project/AndersenCorporation/AndersenWindows/AndersenWindows/Brian-Test/photo-1499970037680-1e5abf25cde2.jpg?h=1200&iar=0&w=1200&hash=B79DA403F104949D9DF11A806C4E26BC',
        alt: 'Window',
        width: '1200',
        height: '1200',
      },
    },
    cta1Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: { Value: { value: 'arrow' } },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: { Value: { value: 'primary' } },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta1Link: {
      value: {
        href: '/',
        text: 'Explore all windows',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta1Modal: null,
    cta1ModalLinkText: { value: '' },
    cta2Style: {
      id: '49a23327-0397-4cce-a930-e76918d37c42',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
      name: 'Primary',
      displayName: 'Primary',
      fields: { Value: { value: 'primary' } },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Icon: {
      id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
      url: 'https://cm.local.ew.andersencorp.com/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
      name: 'Arrow',
      displayName: 'Arrow',
      fields: { Value: { value: 'arrow' } },
      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
      templateName: 'Enum',
    },
    cta2Link: {
      value: {
        href: '/',
        text: 'Explore all doors',
        anchor: '',
        linktype: 'internal',
        class: '',
        title: '',
        target: '',
        querystring: '',
        id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
      },
    },
    cta2Modal: null,
    cta2ModalLinkText: { value: '' },
    eventName: { value: '' },
    eventType: { value: '' },
    eventZone: { value: '' },
  },
};

export default defaultData;
