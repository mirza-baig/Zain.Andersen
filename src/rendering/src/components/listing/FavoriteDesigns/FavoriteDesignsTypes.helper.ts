export interface DesignProps {
  url: string;
  seriesCategory: string;
  interiorImage: InteriorImage;
  exteriorImage: ExteriorImage;
  selections: Selection[];
  shareUrl: string;
  productDetailUrl: string;
  requestAQuoteUrl: string;
  findADealerUrl: string;
  createdDate: string;
  productName?: string;
  productShortDescription: string;
}

export interface InteriorImage {
  src: string;
  alt: string;
}

export interface ExteriorImage {
  src: string;
  alt: string;
}

export interface Selection {
  title: string;
  value: string;
}

declare global {
  interface Window {
    a2a_config?: {
      onclick: number;
      num_services: number;
      exclude_services: Array<string>;
      templates: {
        email: {
          subject: string;
          body: string;
        };
        sms: {
          body: string;
        };
      };
      callbacks: [];
    };
  }
}
