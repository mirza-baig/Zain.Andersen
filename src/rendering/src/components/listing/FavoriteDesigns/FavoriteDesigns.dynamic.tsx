// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { withDatasourceCheck, useSitecoreContext } from '@sitecore-jss/sitecore-jss-nextjs';
import useSWR from 'swr';
import ReactToPrint from 'react-to-print';
// Components
import { Component } from 'src/helpers/Component';
import { getLocalStorageItem } from 'lib/utils/client-storage-utils';
import { useEffect, useMemo, useRef, useState } from 'react';
import { Headline } from 'src/helpers/Headline';
import FavoriteDesignCard from './FavoriteDesignCard.helper';
import { useExternalScript } from 'lib/utils/use-external-script';
import { SvgIcon } from 'src/helpers/SvgIcon';
import { DesignProps } from './FavoriteDesignsTypes.helper';
import { Spinner } from 'src/helpers/Spinner';
import { NoResults } from '../FavoriteProducts/NoResults.helper';
import { shareServicesToExclude } from './FavoriteDesigns.helper';

export type FavoriteDesignsProps =
  Feature.EnterpriseWeb.AndersenWindows.Components.Listing.Favorites.FavoriteDesigns;

const FavoriteDesigns = (props: FavoriteDesignsProps) => {
  const [designs, setDesigns] = useState<Array<DesignProps>>([]);
  const componentRef = useRef<HTMLDivElement>(null);

  const externalScript = 'https://static.addtoany.com/menu/page.js';
  const state = useExternalScript(externalScript);

  const { fields } = props;

  const sitecoreContext = useSitecoreContext();

  const _designs = useMemo(
    () => (getLocalStorageItem('aw_favoritedesigns') as Array<DesignProps>) || [],
    []
  );

  const getProductIds = (): Array<string> => {
    const productIds: Array<string> = [];

    _designs.forEach((design) => {
      for (let i = 0; i < design.selections.length; i++) {
        if (design.selections[i].title === 'Product ID#') {
          productIds.push(design.selections[i].value);
          break;
        }
      }
    });
    return productIds;
  };

  // Define the fetcher function for SWR
  const fetcher = async (url: string, productIds: string[], language: string) => {
    if (productIds.length > 0) {
      const response = await fetch(url, {
        method: 'POST',
        body: JSON.stringify({ favoriteProducts: productIds, language }),
        headers: {
          'Content-Type': 'application/json',
        },
      });

      if (!response.ok) {
        throw new Error('Network response was not ok');
      }

      return response.json();
    }
    return [];
  };

  // Use SWR to fetch the favorite products data
  const { data: productsData, error } = useSWR(
    ['/api/enterprise/favorite-products/'],
    (url: string) =>
      fetcher(url, getProductIds(), sitecoreContext.sitecoreContext.language ?? 'en'),
    { shouldRetryOnError: false }
  );

  useEffect(() => {
    if (productsData) {
      const updatedDesigns = [..._designs];

      _designs.forEach((design) => {
        for (let i = 0; i < design.selections.length; i++) {
          if (design.selections[i].title === 'Product ID#') {
            for (let j = 0; j < productsData.length; j++) {
              if (design.selections[i].value === productsData[j].productId.value) {
                design.productName = productsData[j].productName.value;
                design.productShortDescription = productsData[j].productShortDescription.value;
                break;
              }
            }
          }
        }
      });

      setDesigns(updatedDesigns);
    }
  }, [_designs, productsData]);

  useEffect(() => {
    if (window.a2a_config) {
      window.a2a_config.onclick = 1;
      window.a2a_config.num_services = 8;
      window.a2a_config.exclude_services = shareServicesToExclude;
      window.a2a_config.templates = window.a2a_config.templates || {};
      window.a2a_config.callbacks = window.a2a_config.callbacks || [];
      window.a2a_config.templates = {
        email: {
          subject: 'Check out my Favorites from Andersen Windows',
          body: 'Link to my Favorites on AndersenWindows.com ${link}',
        },
        sms: {
          body: 'Link to my Favorites on AndersenWindows.com ${link}',
        },
      };
    }
  }, [state]);

  const getCreatedAtDate = (dateString: string): string => {
    const date = new Date(dateString);
    const locale = 'en-us';
    const weekday = date.toLocaleDateString(locale, { weekday: 'long' });
    const month = date.toLocaleDateString(locale, { month: 'long' });
    const day = date.toLocaleDateString(locale, { day: '2-digit' });
    const year = date.toLocaleDateString(locale, { year: 'numeric' });

    return `Created ${weekday} ${month}, ${day} ${year}`;
  };

  if (!fields) {
    return null;
  }

  return (
    <Component variant="lg" dataComponent="general/favoritedesigns" {...props}>
      <div className="col-span-12">
        <div className="flex justify-between">
          <Headline {...props} />
          {designs.length > 0 && !error && (
            <ReactToPrint
              trigger={() => {
                return (
                  <button
                    id="print-icon"
                    className="hidden cursor-pointer items-center justify-center text-darkprimary ml:flex"
                  >
                    <SvgIcon icon="print" />
                    <span className="ml-xxs">Print</span>
                  </button>
                );
              }}
              content={() => {
                return componentRef.current;
              }}
              documentTitle="Favorite Designs"
              removeAfterPrint
              copyStyles
              bodyClass="aw"
            />
          )}
        </div>
      </div>
      <div className="col-span-12" ref={componentRef}>
        {(!designs || designs.length === 0) && !error && <NoResults fields={fields} />}
        {error && (
          <div className="font-sans text-sm-m font-medium ml:text-m">
            {
              "We're sorry, we couldn't process your request at this time. Please refresh or try again later."
            }
          </div>
        )}
        {!productsData && !error && (
          <div className="loader flex min-h-[40vh] w-full items-center justify-center">
            <Spinner size={48} />
          </div>
        )}
        {designs &&
          designs.length > 0 &&
          designs.map((designData) => {
            return (
              <>
                <span className="font-serif text-small print:hidden">
                  {getCreatedAtDate(designData.createdDate)}
                </span>
                <div className="relative border border-gray">
                  <FavoriteDesignCard {...designData} />
                </div>
              </>
            );
          })}
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<FavoriteDesignsProps>(FavoriteDesigns);
