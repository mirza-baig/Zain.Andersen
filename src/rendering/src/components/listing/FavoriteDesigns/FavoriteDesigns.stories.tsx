// Global
import { Story, Meta } from '@storybook/react';
// Local
import FavoriteDesigns, { FavoriteDesignsProps } from './FavoriteDesigns.dynamic';
import defaultData from './FavoriteDesigns.mock-data';

export default {
  title: 'Listing/FavoriteDesigns',
  component: FavoriteDesigns,
} as Meta;
const Template: Story<FavoriteDesignsProps> = (props) => <FavoriteDesigns {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
