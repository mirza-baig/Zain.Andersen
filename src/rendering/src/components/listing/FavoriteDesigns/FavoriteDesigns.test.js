// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
import 'lib/jest/match-media-mock';
// Local
import FavoriteDesigns from './FavoriteDesigns.dynamic';
import defaultData from './FavoriteDesigns.mock-data';

it('renders correctly', async () => {
  fetch.mockResponseOnce(JSON.stringify({ productData: [] }));

  const component = snapshot(FavoriteDesigns, { componentProps: defaultData });
  await hasDataComponent(component, 'general/favoritedesigns');
});
