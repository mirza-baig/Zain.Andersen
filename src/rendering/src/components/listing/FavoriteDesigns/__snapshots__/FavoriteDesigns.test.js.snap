// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/favoritedesigns"
  id="id1695242607"
>
  <div
    class="my-8 lg:mx-auto md:max-w-screen-lg theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <div
          class="flex justify-between"
        >
          <div
            class="items-top font-sans [&_a:hover]:underline text-theme-text text-sm-m md:text-m font-bold mb-s"
          >
            <h3>
              Favorite Designs
            </h3>
          </div>
        </div>
      </div>
      <div
        class="col-span-12"
      >
        <div>
          <div>
            <div
              class="font-sans text-m font-medium"
            >
              Looks like you have no saved designs.
            </div>
            <div
              class="body-copy"
            >
              <div
                class="mt-[20px]"
              >
                Let's favorite some things.
              </div>
            </div>
          </div>
          <div
            class="mt-m grid grid-cols-1 gap-4 ml:grid-cols-2"
          >
            <div>
              <div
                class="relative hidden md:block aspect-picture   "
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="591*591"
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="relative block w-full md:hidden"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="Big-Doors-900x900"
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="mt-s font-sans text-s font-heavy"
              >
                Lorem ipsum windows
              </div>
              <div
                class="font-sans text-xs font-medium"
              >
                Subheading text
              </div>
              <div
                class="mt-m mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  aria-label="Explore all windows"
                  class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
                  data-gtm-click=""
                  data-gtm-dl-event="cta_click"
                  href="/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Explore all windows
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
            <div>
              <div
                class="relative hidden md:block aspect-picture   "
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="Window"
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="relative block w-full md:hidden"
              >
                <span
                  style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
                  />
                  <img
                    alt="Window"
                    data-nimg="responsive"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain; object-position: center;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="mt-s font-sans text-s font-heavy"
              >
                Lorem ipsum doors
              </div>
              <div
                class="font-sans text-xs font-medium"
              >
                Subheading text
              </div>
              <div
                class="mt-m mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  aria-label="Explore all doors"
                  class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed"
                  data-gtm-click=""
                  data-gtm-dl-event="cta_click"
                  href="/"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Explore all doors
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="loader flex min-h-[40vh] w-full items-center justify-center"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 48px; height: 48px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
          >
            <img
              alt="loading spinner"
              class="animate-spin"
              data-nimg="fixed"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
      </div>
    </div>
  </div>
</section>
`;
